library/Core/src/Juvix/Core/IR/Typechecker.hs

```diff
@@ -1,434 +1,42 @@
+-- | This file contains the functions and aux functions to typecheck
+-- datatype and function declarations.
+-- Datatype declarations are typechecked by @checkDataType@ in CheckDataType.hs.
+-- Function declarations are typechecked by @typeCheckFuns@ in CheckFunction.hs.
+-- Typechecked declarations are added to the signature.
 module Juvix.Core.IR.Typechecker
   ( module Juvix.Core.IR.Typechecker,
     module Typed,
     module Env,
   )
 where
 
-import qualified Data.HashMap.Strict as HashMap
-import qualified Data.IntMap.Strict as IntMap
-import Data.List.NonEmpty ((<|))
-import qualified Juvix.Core.Application as App
-import qualified Juvix.Core.IR.Evaluator as Eval
+-- import qualified Data.IntMap.Strict as IntMap
+-- import Data.List.NonEmpty ((<|))
+-- import qualified Juvix.Core.Application as App
+-- import qualified Juvix.Core.IR.CheckTerm as IR
+-- import qualified Juvix.Core.IR.Evaluator as Eval
 import Juvix.Core.IR.Typechecker.Env as Env
 import Juvix.Core.IR.Typechecker.Types as Typed
 import qualified Juvix.Core.IR.Types as IR
 import qualified Juvix.Core.IR.Types.Base as IR
-import qualified Juvix.Core.Parameterisation as Param
+-- import qualified Juvix.Core.Parameterisation as Param
```

Comment:
Can these commented imports be removed now?

Reply:
Yup. Done.